libyul/backends/evm/OptimizedEVMCodeTransform.cpp

```diff
@@ -0,0 +1,433 @@
+				deficit,
+				msg
+			));
+
+			m_assembly.markAsInvalid();
+		}
+		else
+			m_assembly.appendInstruction(evmasm::swapInstruction(_i));
+	}, [&](StackSlot const& _slot) {
+		if (auto depth = util::findOffset(m_stack | ranges::views::reverse, _slot))
+		{
+			if (*depth < 16)
+			{
+				m_assembly.appendInstruction(evmasm::dupInstruction(static_cast<unsigned>(*depth + 1)));
+				return;
+			}
+			else if (!canBeFreelyGenerated(_slot))
+			{
+				int deficit = static_cast<int>(*depth - 15);
+				YulString varName = slotVariableName(_slot);
+				string msg = (varName.empty() ? "Slot " + stackSlotToString(_slot) : "Variable " + varName.str())
+					+ " is " + to_string(*depth - 15) + " too deep in the stack " + stackToString(m_stack);
+				m_stackErrors.emplace_back(StackTooDeepError(
+					m_currentFunctionInfo ? m_currentFunctionInfo->function.name : YulString{},
+					varName,
+					deficit,
+					msg
+				));
+				m_assembly.markAsInvalid();
+				m_assembly.appendConstant(u256(0xDEADBEEF));
+				return;
+			}
+		}
+		yulAssert(static_cast<int>(m_stack.size()) == m_assembly.stackHeight(), "");
+		std::visit(util::GenericVisitor{
+			[&](LiteralSlot const& _literal)
+			{
+				m_assembly.setSourceLocation(locationOf(_literal));
+				m_assembly.appendConstant(_literal.value);
+			},
+			[&](FunctionReturnLabelSlot const&)
+			{
+				yulAssert(false, "Cannot produce function return label.");
+			},
+			[&](FunctionCallReturnLabelSlot const& _returnLabel)
+			{
+				if (!m_returnLabels.count(&_returnLabel.call.get()))
+					m_returnLabels[&_returnLabel.call.get()] = m_assembly.newLabelId();
+				m_assembly.appendLabelReference(m_returnLabels.at(&_returnLabel.call.get()));
+			},
+			[&](VariableSlot const& _variable)
+			{
+				if (m_currentFunctionInfo)
+					if (util::contains(m_currentFunctionInfo->returnVariables, _variable))
+					{
+						m_assembly.appendConstant(0);
+						return;
+					}
+				yulAssert(false, "Variable not found on stack.");
+			},
+			[&](TemporarySlot const&)
+			{
+				yulAssert(false, "Function call result requested, but not found on stack.");
+			},
+			[&](JunkSlot const&)
+			{
+				// Note: this will always be popped, so we can push anything.
+				// TODO: discuss if PC is in fact a good choice here.
+				// Advantages:
+				// - costs only 2 gas
+				// - deterministic value (in case it is in fact used due to some bug)
+				// - hard to exploit in case of a bug
+				// - distinctive, since it is not generated elsewhere
+				// Disadvantages:
+				// - static analysis might get confused until it realizes that these are always popped
+				// Alternatives:
+				// - any other opcode with cost 2
+				// - unless the stack is empty: DUP1
+				// - the constant 0
+				// Note: it might even make sense to introduce a specific assembly item for this, s.t.
+				//       the peephole optimizer can deal with this (e.g. POP PUSHJUNK can be removed).
+				m_assembly.appendInstruction(evmasm::Instruction::PC);
```

Comment:
```suggestion
				m_assembly.appendInstruction(evmasm::Instruction::CODESIZE);
```
may be a good alternative.